client/src/components/PostComment.tsx

```diff
@@ -22,23 +22,36 @@ import {
 } from "@chakra-ui/react";
 import { BiTrash, BiEdit } from "react-icons/bi";
 import { useNavigate } from "react-router-dom";
-import { useRef } from "react";
+import { useRef, useEffect, useState } from "react";
 import { useSelector } from "react-redux";
-
 import { userSelector } from "redux/user";
 import { genericErrorHandler } from "utils";
-
 import type { Comment } from "types";
 
 const API_URL = `${import.meta.env.VITE_API_SERVER_URL}/api/v1`;
 
 const PostComment = (props: { comment: Comment }) => {
-  const { isOpen, onOpen, onClose } = useDisclosure();
   const { user } = useSelector(userSelector);
+  const AuthToken = user?.bearerToken;
```

Comment:
```suggestion
  const AuthToken = user?.bearerToken || "";
```
Need to guard against `undefined`